### ISSUE TITLE
Wrong nDCG result

### ISSUE THREAD

**Issue by dionet** [lenskit/lenskit#869]
When measuring nDCG in LensKit 3, the generated IDCG list length doesn't comply with the list size set using RecommendEvalTask.setListSize(). This means the calculated nDCG value will be lower than it actually is.

You can reproduce this bug using the following file: [TestMetrics.txt](https://github.com/lenskit/lenskit/files/138352/TestMetrics.txt)

Result:
```
RMSE.ByRating: 0.0
TopN.nDCG @ 10: 0.39331593152088673
TopN.nDCG @ Inf: 1.0
```

**Comment by mdekstrand** [comment]
@dionet This should be fixed in #873.

**Comment by dionet** [comment]
@mdekstrand Thanks! The fix works fine.
